### ISSUE TITLE
ModuleNotFoundError: No module named 'maggy.experiment_config'

### ISSUE THREAD

**Issue by crakama** [logicalclocks/maggy#108]
I have can install maggy on my PySpark cluster from pip but whenever I issue this command `from maggy.experiment_config import OptimizationConfig` I get the error `ModuleNotFoundError: No module named 'maggy.experiment_config'`.  Any idea of what could be happening. I am using JupiterLab with Python3 Kenrnel

`from maggy import experiment` says no module named `hops`.

The imports that I have seen working are from `maggy.ablation import AblationStudy` and `from maggy import Searchspace`

Edit:
I noticed that I cannot use pre-release version (if it has a fix of this that is). I get this error when I try to install the pre-release version
_**ERROR: Could not find a version that satisfies the requirement maggy==1.0.0rc0 (from versions: 0.0.1, 0.1, 0.1.1, 0.2, 0.2.1, 0.2.2, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.4.0, 0.4.1, 0.4.2, 0.5.0, 0.5.1, 0.5.2, 0.5.3)
ERROR: No matching distribution found for maggy==1.0.0rc0**_

**Comment by RiccardoGrigoletto** [comment]
Hi Catherine, did you try to download the wheel from https://github.com/logicalclocks/maggy/releases/tag/1.0.0rc0 and install it?

**Comment by crakama** [comment]
I think it is also important to mention somewhere in the documentation the prerequisites e.g that the library requires Python>=3.7 . This is because in some cases updating a package is not as easy as downloading python and installing it, it a whole process of sending emails to ops teams and waiting for the request to be added in a sprint and things like that, and this can save up some time to get started with the actual work :-).

**Comment by moritzmeister** [comment]
Hey @crakama,
which version of pip are you using? Are you using any proxy for pip?
I tried reproducing your installation error but it works as expected with pip 21.1.1. 

[The requirement of Python>=3.7 is among the PyPi classifiers](https://github.com/logicalclocks/maggy/blob/f71510d5c7e54503456041392748240d95ad9424/setup.py#L72).

**Comment by crakama** [comment]
I upgraded my ipython kernel in jupiterlab  from 3.6 to 3.8 and I could install maggy 0.5.3 but  seamlessly graded the version
 using `pip install --user --pre --upgrade --no-deps --force-reinstall maggy`

I am getting other errors not related to maggy but once the errors are fixed I will test maggy's source code again and see if the problem is solved

Otherwise if a python kernel of **below** 3.7 is used, the package 0.5.3 gets installed and then you will be required to manually install other other dependency packages such as `ERROR: Could not find a version that satisfies the requirement scikit-optimize==0.7.4 (from maggy) (from versions: 0.1, 0.2, 0.3, 0.4rc1, 0.4, 0.5, 0.5.1, 0.5.2, 0.7rc4, 0.7, 0.7.1, 0.7.2)
ERROR: No matching distribution found for scikit-optimize==0.7.4`
Even after manual installations of all dependencies, the error that maggy requires python 3.7 and above will still show up.

**Comment by moritzmeister** [comment]
Yes like I said, [Python 3.7 is a requirement](https://github.com/logicalclocks/maggy/blob/f71510d5c7e54503456041392748240d95ad9424/setup.py#L66).

And like I said please make sure to use an up to date version of pip.
